config/routes.rb

```diff
@@ -13,23 +13,23 @@
   get 'tasks/organisation', to: 'tasks#organisation'
 
   resources :datasets do
-    member do
-      match 'new/licence',   to: 'datasets#licence',   via: [:get, :post]
-      match 'new/location',  to: 'datasets#location',  via: [:get, :post]
-      match 'new/frequency', to: 'datasets#frequency', via: [:get, :post]
-      match 'new/addfile',   to: 'datasets#addfile',   via: [:get, :post]
-      match 'new/files',     to: 'datasets#files',   via: [:get, :post]
-      match 'new/adddoc',    to: 'datasets#adddoc',    via: [:get, :post]
-      match 'new/documents', to: 'datasets#documents',   via: [:get, :post]
-
-      match 'edit/licence',   to: 'datasets#edit_licence',   via: [:get, :put]
-      match 'edit/location',  to: 'datasets#edit_location',  via: [:get, :put]
-      match 'edit/frequency', to: 'datasets#edit_frequency', via: [:get, :put]
-      match 'edit/addfile',   to: 'datasets#edit_addfile',   via: [:get, :put]
-      match 'edit/adddoc',    to: 'datasets#edit_adddoc',    via: [:get, :put]
-
-      match 'publish',       to: 'datasets#publish',   via: [:get, :post]
+    get 'show/:id', to: 'datatsets#show'
 
+    member do
+      match 'licence',       to: 'datasets#licence',     via: [:post, :put]
+      match 'location',      to: 'datasets#location',    via: [:post, :put]
+      match 'frequency',     to: 'datasets#frequency',   via: [:post, :put]
+
+      resources :files,     controller: 'datafiles', param: :file_id
+      resources :documents, controller: 'datafiles', param: :file_id
+
+      get 'new/licence',    to: 'datasets#licence'
```

Comment:
Not a big fan of new/ and edit/ although I know it makes it easier to determine whether you're editing or not, but I guess it doesn't matter too much.  Maybe we can discuss it next time we're together to see if it introduces more complexity or less.  Just interested in what the idiomatic ruby way is.